src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -744,51 +842,68 @@ await ExecuteWithFailOverPolicyAsync(
+            IEnumerable<KeyValueSelector> selectors,
+            CancellationToken cancellationToken)
         {
-            var serverData = new Dictionary<string, ConfigurationSetting>(StringComparer.OrdinalIgnoreCase);
-
-            // Use default query if there are no key-values specified for use other than the feature flags
-            bool useDefaultQuery = !_options.KeyValueSelectors.Any(selector => selector.KeyFilter == null ||
-                !selector.KeyFilter.StartsWith(FeatureManagementConstants.FeatureFlagMarker));
+            Dictionary<string, ConfigurationSetting> data = new Dictionary<string, ConfigurationSetting>();
 
-            if (useDefaultQuery)
+            foreach (KeyValueSelector loadOption in selectors)
             {
-                // Load all key-values with the null label.
-                var selector = new SettingSelector
-                {
-                    KeyFilter = KeyFilter.Any,
-                    LabelFilter = LabelFilter.Null
-                };
-
-                await CallWithRequestTracing(async () =>
+                if (string.IsNullOrEmpty(loadOption.SnapshotName))
                 {
-                    await foreach (ConfigurationSetting setting in client.GetConfigurationSettingsAsync(selector, cancellationToken).ConfigureAwait(false))
+                    var selector = new SettingSelector()
                     {
-                        serverData[setting.Key] = setting;
-                    }
-                }).ConfigureAwait(false);
-            }
+                        KeyFilter = loadOption.KeyFilter,
+                        LabelFilter = loadOption.LabelFilter
+                    };
 
-            foreach (KeyValueSelector loadOption in _options.KeyValueSelectors)
-            {
-                IAsyncEnumerable<ConfigurationSetting> settingsEnumerable;
+                    var matchConditions = new List<MatchConditions>();
 
-                if (string.IsNullOrEmpty(loadOption.SnapshotName))
-                {
-                    settingsEnumerable = client.GetConfigurationSettingsAsync(
-                        new SettingSelector
+                    await CallWithRequestTracing(async () =>
+                    {
+                        AsyncPageable<ConfigurationSetting> pageableSettings = client.GetConfigurationSettingsAsync(selector, cancellationToken);
+
+                        await foreach (Page<ConfigurationSetting> page in _options.PageableManager.GetPages(pageableSettings).ConfigureAwait(false))
                         {
-                            KeyFilter = loadOption.KeyFilter,
-                            LabelFilter = loadOption.LabelFilter
-                        },
-                        cancellationToken);
+                            // Server etag will never be null or empty because non-conditional List key-values API always returns 200
+                            ETag serverEtag = (ETag)page?.GetRawResponse()?.Headers.ETag;
```

Comment:
If we've gotten to this line, we have retrieved page(s) without exception, meaning there will be a page, and there will be a raw response with headers. There is no need for null checks here. Additionally, the `RequestFailedException` below doesn't appear needed given our API contract is to return a server etag. Unless the consideration here is that the user may be using an older API version?

Reply:
I wasn't really able to find the API requirements before, so I was convinced that it made sense to add null checks as a precaution, but it makes sense to remove it.